File: 27October2025/Java/nodes.java
public class Node {
    public String data; //payload stored in the node
    public Node next;  //Reference to the next node
    public Node(String data) {
        this.data = data;
        this.next = null;
    }

    public class LinkedList {
        public Node head;
        public LinkedList() {
            this.head = null;
        }

        public addElement(String data) {
            Node new_node = new Node(data);
            new_node.next = this.head;
            this.head = new_node;
        }
    }

    public int listlength; {
        //μετρητής
        int counter = 0;
        Node current = this.head;

        while (current !=null) {
            counter++;
            current = current.next;
        }
        return counter;
    }
}

File: techpro2710/nodes.py


def length(self):
    counter = 1
    current = self.head
    if current == None:
        return 0
    
    while current.next != None:
        current = current.next
        counter = counter + 1

def printEverything(self):
    current = self.head
    if current is None:
        return
    while current.next != None:
        print(current.data)
        current = current.next


my_linked_list.addElement("George")
my_linked_list.addElement("Afroxilanthi")

print(my_linked_list.length())